lib/appoptics_apm/inst/bunny-consumer.rb

```diff
@@ -9,59 +9,57 @@ def self.included(klass)
-          if mp.respond_to?(controller_key)
-            value = mp.method(controller_key).call
-            kvs[:Controller] = value if value
-          end
-
-          action_key = AppOpticsAPM::Config[:bunnyconsumer][:action]
-          if mp.respond_to?(action_key)
-            value = mp.method(action_key).call
-            kvs[:Action] = value if value
-          end
-
-          if kvs[:Queue]
-            kvs[:URL] = "/bunny/#{kvs[:Queue]}"
-          else
-            kvs[:URL] = "/bunny/consumer"
-          end
-
-          if AppOpticsAPM::Config[:bunnyconsumer][:log_args] && @arguments
-            kvs[:Args] = @arguments.to_s
-          end
-
-          kvs[:Backtrace] = AppOpticsAPM::API.backtrace if AppOpticsAPM::Config[:bunnyconsumer][:collect_backtraces]
-
-          kvs
-        rescue => e
-          AppOpticsAPM.logger.debug "[appoptics_apm/debug] #{__method__}:#{File.basename(__FILE__)}:#{__LINE__}: #{e.message}" if AppOpticsAPM::Config[:verbose]
-        ensure
-          return kvs
+        kvs = {}
+        kvs[:Spec] = :job
+        kvs[:Flavor] = :rabbitmq
+        kvs[:RemoteHost]  = @channel.connection.host
+        kvs[:RemotePort]  = @channel.connection.port.to_i
+        kvs[:VirtualHost] = @channel.connection.vhost
+
+        mp = args[1]
+        kvs[:RoutingKey] = args[0].routing_key if args[0].routing_key
+        kvs[:MsgID]      = args[1].message_id  if mp.message_id
+        kvs[:AppID]      = args[1].app_id      if mp.app_id
+        kvs[:Priority]   = args[1].priority    if mp.priority
+
+        if @queue.respond_to?(:name)
+          kvs[:Queue] = @queue.name
+        else
+          kvs[:Queue] = @queue
         end
+
+        # Report configurable Controller/Action KVs
+        # See AppOpticsAPM::Config[:bunnyconsumer] in lib/appoptics_apm/config.rb
+        # Used for dashboard trace filtering
+        controller_key = AppOpticsAPM::Config[:bunnyconsumer][:controller]
+        if mp.respond_to?(controller_key)
+          value = mp.method(controller_key).call
+          kvs[:Controller] = value if value
+        end
+
+        action_key = AppOpticsAPM::Config[:bunnyconsumer][:action]
+        if mp.respond_to?(action_key)
+          value = mp.method(action_key).call
+          kvs[:Action] = value if value
+        end
+
+        if kvs[:Queue]
+          kvs[:URL] = "/bunny/#{kvs[:Queue]}"
+        else
+          kvs[:URL] = "/bunny/consumer"
+        end
+
+        if AppOpticsAPM::Config[:bunnyconsumer][:log_args] && @arguments
+          kvs[:Args] = @arguments.to_s
+        end
+
+        kvs[:Backtrace] = AppOpticsAPM::API.backtrace if AppOpticsAPM::Config[:bunnyconsumer][:collect_backtraces]
+
+        kvs
+      rescue => e
+        AppOpticsAPM.logger.debug "[appoptics_apm/debug] #{__method__}:#{File.basename(__FILE__)}:#{__LINE__}: #{e.message}" if AppOpticsAPM::Config[:verbose]
+      ensure
+        return kvs
```

Comment:
why take out (or why was it once there) the begin/end?

Reply:
I took it out because it is not needed, the def-end block is sufficient. Don't know why it was there.